### PR TITLE
[2400] Add available modern languages to edit options

### PR DESCRIPTION
**Pull request by faissalMT**
### Context
As it stands the subject editing functionality cannot add or modify modern languages. In order to be able to modify the list of modern languages the frontend will need to be provided with the modern languages.

### Changes proposed in this pull request
In edit options add a subsection for potential modern languages that can be chosen from that is nil for any level that is not secondary, as other levels cannot have modern languages.

### Guidance to review

* argue about naming for a week or so

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by faissalMT** [comment]
[[S] Add potential "Modern Language" subjects to edit options](https://trello.com/c/MM7h4rT9/2400-s-add-potential-modern-language-subjects-to-edit-options)

File: app/models/concerns/courses/edit_options/subjects_concern.rb [hunk 1]
```diff
@@ -13,6 +13,15 @@ def potential_subjects
             class: CourseSerializersService.new.execute,
           )[:data]
         end
+
+        def potential_modern_languages
```
Comment (defong) [inline app/models/concerns/courses/edit_options/subjects_concern.rb]: shouldn't it be just be ` modern_languages_subject`

Comment (faissalMT) [reply app/models/concerns/courses/edit_options/subjects_concern.rb]: I feel like this is more descriptive of its function

File: app/models/concerns/courses/edit_options.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ def edit_course_options
           show_start_date: show_start_date?,
           show_applications_open: show_applications_open?,
           subjects: potential_subjects,
+          potential_modern_languages: potential_modern_languages,
```
Comment (defong) [inline app/models/concerns/courses/edit_options.rb]: I meant here

File: app/models/concerns/courses/edit_options/subjects_concern.rb (hunk 1 above)
Comment (defong) [reply app/models/concerns/courses/edit_options/subjects_concern.rb]: https://github.com/DFE-Digital/manage-courses-backend/pull/926#discussion_r338504345

File: app/models/concerns/courses/edit_options.rb (hunk 2 above)
Comment (faissalMT) [reply app/models/concerns/courses/edit_options.rb]: Same as my previous comment

Comment (defong) [reply app/models/concerns/courses/edit_options.rb]: i disagree...you either have it or not have but not potential having it not having it

**Comment by davidgisbey** [comment]
Looks good to me. I think it's fairly clear that potential_modern_languages contains a list of subjects to select from. No point arguing semantics.

File: app/models/concerns/courses/edit_options.rb (hunk 2 above)
Comment (faissalMT) [reply app/models/concerns/courses/edit_options.rb]: These are intended to be the modern languages a user may be able to select from if any.

**Comment by defong** [comment]
> Looks good to me. I think it's fairly clear that potential_modern_languages contains a list of modern language subjects to select from. No point arguing semantics.

```
subjects: potential_subjects,
potential_modern_languages: potential_modern_languages
```

the exacting line above is called `subject` not `potential_subjects`
so why does `potential_modern_languages` is `potential_modern_languages`?

> These are intended to be the modern languages a user may be able to select from if any.


`semantics` are important
a server gives `exactly` what it can offer not `potential` what it can offer
a server gives `exactly` an empty list not `potential` empty list

the client can select  a `potential` item of their interest

this is client semantic leaking into server

File: app/models/concerns/courses/edit_options.rb (hunk 2 above)
Comment (misaka) [reply app/models/concerns/courses/edit_options.rb]: I think I agree about making the naming convention standard, and removing `potential` from the key, here. I would go one step further and favour the language `available` over `potential`, but that's less important.

```suggestion
          modern_languages: available_modern_languages,
```

**Comment by timabell** [comment]
> No point arguing semantics

I disagree.

there-are-two-hard-problems-in-computer-science-0-naming-26556167

What about `available_modern_languages` ?

**Comment by defong** [comment]
> @defong
> 
> > in full context it just looks potential ...
> 
> what are you on about? confused



here's the full context annotated fot you @timabell 
```javascript
{
    //meta of something
  "meta": {
     //edit_options of something
    "edit_options":{
       //name of something
      "entry_requirements": [...],
      //name of something
      "qualifications": [...],
      //name of something
      "age_range_in_years": [...],
      //name of something
      "start_dates": [...],
      //name of something
      "study_modes": [...],
      //name of something
      "show_is_send": [...],
      //name of something
      "show_start_date": [...],
      //name of something
      "show_applications_open": [...],
      //name of something
      "subjects": [...],  
      //potential name of something
      "potential_modern_languages":  [...]  
    }
  }
}
```

**Comment by timabell** [comment]
I'm taking over finishing this off

**Comment by timabell** [comment]
rebased

**Pull request merged**